app/assets/javascripts/init_handin_datetimepickers.js

```diff
@@ -0,0 +1,39 @@
+// Initialize all Flatpicker Datetime Pickers on the page
+
+$(document).ready(function() {
+
+  function createDatePicker(selector, config) {   
+    var elt = $(selector)
+    var defaults = {
+      enableTime: true,
+      altInput: true,
+      defaultDate: new Date(moment(elt.val()))
+    }
+    Object.assign(defaults, config);
+    /* Invoke flatpickr library */
+    return flatpickr(selector, defaults)
+  }
+
+  /* Create all 3 date pickers */
+  createDatePicker('#assessment_start_at')
+  var end_at_pickr = createDatePicker('#assessment_end_at')
+  var grading_deadline_pickr = createDatePicker('#assessment_grading_deadline')
+
+  function onCloseHandler(selected_dates, date_str, flatpickr_inst) {
+      var cur_date = selected_dates[0]
+      if (grading_deadline_pickr.selectedDates[0].getTime() < cur_date.getTime()) {
+        grading_deadline_pickr.setDate(cur_date, true);
+      }
+
+      if (end_at_pickr.selectedDates[0].getTime() < cur_date.getTime()) {
+        end_at_pickr.setDate(cur_date, true);
+      }
+
+      // Reinstall handler
+      flatpickr_inst.set('onClose', [onCloseHandler])
```

Comment:
What browser did you try it on? I tried on Chrome without this and it wouldn't work multiple times